### ISSUE TITLE
Refresh token does no work.

### ISSUE THREAD

**Issue by plabadie** [SAP/cloud-commerce-spartacus-storefront#4392]
Backend version: 1905.0 public release.
Spartacus version: develop branch on 2019-09-03

Steps to reproduce:

- Set the following configs on the backend's local.properties file:

```
oauth2.accessTokenValiditySeconds=45
oauth2.refreshTokenValiditySeconds=300
```

- Restart the commerce backend (the token configs don't seem to apply unless the instance is restarted, it isn't enough to add them via hac)"

- Sign in with a customer.
I got this king of response for the authentication:
```
{
  "access_token" : "14399ab4-2e63-4ee2-a0a8-3ff493e09142",
  "token_type" : "bearer",
  "refresh_token" : "5c28d5b1-c978-4e38-adfa-c1e41ef97b9c",
  "expires_in" : 13,
  "scope" : "basic openid"
}
```

Expected result:
This should expire a user token after 45 seconds and allow for a refresh token to be obtained and be valid for 300 seconds.
See also the original implementation ticket testing instructions: https://jira.hybris.com/browse/SPA-312?focusedCommentId=721125&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-721125

Actual results:
The token does expire after 45 seconds, but no refresh token can be obtained.  The backend's response is:
```
{
  "error" : "invalid_grant",
  "error_description" : "Invalid refresh token: 5c28d5b1-c978-4e38-adfa-c1e41ef97b9c"
}
```

--- 
Note this may be a backend issue, but there is investigation required to see if there is any issue on the frontend or if it's a backend issue.

**Comment by plabadie** [comment]
The issue is fixed in PR #4443 

refresh requests can only be performed once and s switchMap was to blame for cancelling the refresh request and the other concurrent refresh requests failed.

**Comment by plabadie** [comment]
The issue is fixed in PR #4443 

refresh requests can only be performed once and s switchMap was to blame for cancelling the refresh request and the other concurrent refresh requests failed.
